### PR TITLE
[WordPress] updated ignored list

### PR DESCRIPTION
**Pull request by rohanvakharia**
**Reasons for making this change:**

To ignore everything except the plugins and theme

**Links to documentation supporting these rule changes:**

https://core.trac.wordpress.org/browser/trunk/.gitignore

If this is a new template:

https://wordpress.org

### CODE REVIEW

File: WordPress.gitignore [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
-/.htaccess
-/license.txt
-/readme.html
-/sitemap.xml
-/sitemap.xml.gz
+# ignore specific themes
+wp-content/themes/twenty*/
 
+# ignore node dependency directories
+node_modules/
```
Comment (malithmcr) [inline WordPress.gitignore]: Does Wordpress use node_modules in the root?

Comment (rohanvakharia) [reply WordPress.gitignore]: Yes. 
Here you will find the list of packages using by WordPress:
https://core.trac.wordpress.org/browser/trunk/package.json

**Comment by shiftkey** [comment]
@rohanvakharia thanks!

**Comment by kolorafa** [comment]
How does those rules ignore all files in root directory?
https://github.com/github/gitignore/blob/499ae899e7b54e701e878759f73d9092302fd07a/WordPress.gitignore#L1-L2

Isn't there something missing like /*  ?
Current (after merge) gitignore includes wp-config:

```
$ git add . --dry-run|grep wp-config
add 'wp-config-sample.php'
add 'wp-config.php'
```

**Comment by hdwebpros** [comment]
I'm not sure why this PR was merged. It's confusing. 

If I see `# ignore everything in the root except the "wp-content" directory. ` then I shouldn't see `# ignore log files and databases` since those are in the root.

Also, like @kolorafa mentioned, it seems that there is something missing. We've used:

```
# Ignore everything, but all to descend into subdirectories
*
!*/
```
To ignore everything, but allow the including of what you want included. So, you could have

```
# "themes/my-theme" directory
!wp-content/themes/my-theme/**
```
Later in your gitignore file and have that work fine.

**Comment by pedrosanta** [comment]
It's just me or this merge is a utter and complete _trainwreck_? I've got all seed projects broken, basically allowing me to add/stage pretty much of a complete WP install into the commit.

This wasn't nowhere near as tested once by the submitter.

**Pull request merged**